File: dypro_leetcode/dia 019/maxProfit.js
/**
 * @param {number[]} prices
 * @return {number}
 */
var maxProfit = function(prices) {
    let l = 0;
    let r = 0;
    let max = 0;
    while (r < prices.length){
        r++;   
        a = prices[r] - prices[l];
        if (a > max) max = a;
        if (prices[r] < prices[l]) l = r;
    }
    return max;
};

File: leetcode/dia 050/maxProfit.cpp
class Solution {
public:
    int maxProfit(vector<int>& prices) {
        int max = 0, l = INT_MAX;
        for (int n : prices){
            if (n - l > max) max = n - l;
            if (n < l) l = n;
        }
        return max;
    }
};